File: SapphireYours_Common/src/grafl/sy/buttons/Button.java
package grafl.sy.buttons;

import grafl.sy.game.Game;
import grafl.sy.renderer.VectorRenderer;

public class Button 
{
	Game game;
	
	float x;
	float y;
	float width;
	float height;
	Runnable triggeraction;
	
	boolean isPressed;
	
	
	public Button(Game game, float x, float y, float width, float height, Runnable triggeraction)
	{
		this.game = game;
		this.x = x;
		this.y = y;
		this.width = width;
		this.height = height;
		this.triggeraction = triggeraction;
		this.isPressed = false;
	}
	
	// need to be implemented - graphics appearance
	public void draw(VectorRenderer vr)	
	{	        	    	
	}
	
	public void setPosition(float x, float y)
	{	
		this.x = x;
		this.y = y;
	}
	
	// ------------------- handle touch input commands ------------
	public boolean onPointerDown(int x, int y)
	{
		if (x>=this.x && x<this.x+width && y>=this.y && y<this.y+height)
		{	isPressed = true;
			return true;		
		}
		return false;
	}
	
    public void onPointerUp()
	{
		if (isPressed)
		{	isPressed = false;
			triggeraction.run();
		}
	}
		
	public void onPointerMove(int x, int y)
	{
		if (!isPressed)
		{	return;
		}
		isPressed = (x>=this.x && x<this.x+width && y>=this.y && y<this.y+height);
	}



}

File: game/src/button/Button.js

var Button = function()
{
    this.game = null;    
    this.x = 0.0;
    this.y = 0.0;
    this.width = 0.0;
    this.height = 0.0;
    this.triggeraction = null;    
    this.isPressed = false;
};

Button.prototype.$ = function(game, x, y, width, height, triggeraction)
{
    this.game = game;
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
    this.triggeraction = triggeraction;
    this.isPressed = false;
};
    
Button.prototype.draw = function(vr) 
{                       
};
    
Button.prototype.setPosition = function(x,y)
{   
    this.x = x;
    this.y = y;
};
    
    // ------------------- handle touch input commands ------------
Button.prototype.onPointerDown = function(x, y)
{
    if (x>=this.x && x<this.x+this.width && y>=this.y && y<this.y+this.height)
    {   this.isPressed = true;
        return true;        
    }
    return false;
};
    
Button.prototype.onPointerUp = function()
{
    if (this.isPressed)
    {   this.isPressed = false;
        this.triggeraction.run();
    }
};
        
Button.prototype.onPointerMove = function(x, y)
{
    if (!this.isPressed)
    {   return;
    }
    this.isPressed = (x>=this.x && x<this.x+this.width && y>=this.y && y<this.y+this.height);
};
